### PR TITLE
Create Floating Action Bar component

### PR DESCRIPTION
**Pull request by Chrischuck**
#485 

Added a couple of screen shots below.

Not in spec, but I figured the floating action bar would want to stay in view if the keyboard is open - can be removed if unnecessary. I think there should be a few more options to account for the NavBar at the bottom and what should be considered "bottom"

Either way those shouldn't be too hard to adjust.

**iOS**
Top
Screenshot 2023-02-20 at 2 05 05 PM
bottom w/ keyboard
Screenshot 2023-02-20 at 2 23 05 PM

**Android**
bottom
Screenshot 2023-02-20 at 3 14 13 PM
bottom w/ keyboard
Screenshot 2023-02-20 at 3 01 35 PM

### CODE REVIEW

File: src/components/UiLibrary.js [hunk 1]
```diff
@@ -47,6 +48,22 @@ const UiLibrary = () => {
...
+            mode="contained"
+          />
+          <IconButton
+            icon="plus-sign"
+            mode="contained"
+          />
+          <IconButton
+            icon="icon-fave"
+            mode="contained"
+          />
```
Comment (kueda) [inline src/components/UiLibrary.js:64]: Minor thing but can you

- [ ] add the text "FloatingActionBar" to the FloatingActionBar
- [ ] have one button
- [ ] and make the bottom margin 100px

Those would make it a bit clearer that this is part of the UI Library and not a real part of the app.

**Review by kueda — CHANGES_REQUESTED** [review]
Looks good, just some minor requests. Regarding the keyboard, I think that's good to leave in, but I don't think we'll need it initially (we're going to use this on the Group Photos view, where there shouldn't be a way to open a keyboard, in theory).

File: src/components/SharedComponents/FloatingActionBar.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import { View } from "components/styledComponents";
+import * as React from "react";
+import { useEffect, useState } from "react";
+import { Keyboard, StyleSheet } from "react-native";
+import { useTheme } from "react-native-paper";
+
+type Props = {
+  position: "topStart" | "topEnd" | "bottomStart" | "bottomEnd",
+  marginX: string,
+  marginY: string,
```
[the remark this answers is not in the document]
Comment (Chrischuck) [reply src/components/SharedComponents/FloatingActionBar.js:12]: my intent was also along the lines with what you're thinking, unfortunately you cant dynamically build an associated tailwind class like 'my-[${height}px]'

i do think passing a pixel value is a better way to do things but we'd probably have to use the 'style' prop instead

File: src/components/UiLibrary.js (hunk 1 above)
Comment (Chrischuck) [reply src/components/UiLibrary.js:64]: can do

File: src/components/SharedComponents/FloatingActionBar.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Chrischuck) [reply src/components/SharedComponents/FloatingActionBar.js]: @kueda I just ended up taking away `marginX` and `marginY` props and opted for a `containerClass` prop for more flexibility and to allow as much tailwinds usage as possible.

**Review by kueda — APPROVED** [review]
Looking good. Thanks again!

**Pull request merged**